cmake/CubBuildCompilerTargets.cmake

```diff
@@ -94,7 +94,7 @@ function(cub_build_compiler_targets)
   foreach (cxx_option IN LISTS cxx_compile_options)
     target_compile_options(cub.compiler_interface INTERFACE
       $<$<COMPILE_LANGUAGE:CXX>:${cxx_option}>
-      $<$<COMPILE_LANG_AND_ID:CUDA,NVCXX>:${cxx_option}>
+      $<$<COMPILE_LANG_AND_ID:CUDA,NVCXX>:${cxx_option}> # TODO check that this works with the new nvc++ CMake recipe
```

Comment:
This compiler vendor ID doesn't exist

Reply:
Plus lets use ` $<COMPILE_LANG_AND_ID:CUDA,...>` to be consistent with the rest of the file

Reply:
NVHPC is the new one

Reply:
Updated to use the NVHPC id and `COMPILER_LANG_AND_ID` genex consistently.